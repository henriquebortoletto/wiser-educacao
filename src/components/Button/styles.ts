import styled from "styled-components";

export const Button = styled.button`
	font-size: 16px;
	font-weight: 600;
	text-transform: uppercase;
	background: none;
	border: none;
	background-image: linear-gradient(to right, #383e71, #9d25b0);
	border-radius: 8px;
	color: #fff;
	width: 256px;
	height: 48px;
	cursor: pointer;
	outline: 0;
	box-shadow: 0px 10px 25px #cf99db;
	@media (max-width: 650px) {
		width: 168px;
		position: relative;
		top: calc(-48px / 2);
		left: calc(50% - 84px);
		box-shadow: none;
	}
`;
