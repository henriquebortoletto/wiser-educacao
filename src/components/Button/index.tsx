import * as S from "./styles";
import axios from "../../services/api";

export default function Button() {
  async function handleClick() {
    try {
      // const response = await axios.post("/wiser", {
      //   email: "bortolettohenrique@gmail.com",
      //   password: "1234567",
      // });
      const response = await axios.get("/wiser").then((response) => {
        console.log(response.data);
      });
      console.log(response);
    } catch (error) {
      alert("Erro ao tentar logar! Tente mais tarde");
    }
  }

  return <S.Button onClick={() => handleClick()}>Entrar</S.Button>;
}
