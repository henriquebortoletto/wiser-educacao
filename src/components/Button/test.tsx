import { render } from "@testing-library/react";

import Button from "./index";

describe("<Button />", () => {
	it("should render text", () => {
		const { container } = render(<Button type="submit" label="entrar" />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
