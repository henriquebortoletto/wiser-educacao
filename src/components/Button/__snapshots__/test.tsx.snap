// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render text 1`] = `
<button
  class="styles__Button-sc-11qq8g6-0 bMHdRd"
  type="submit"
>
  entrar
</button>
`;
